src/route-handlers/list-workflows/list-workflows.ts

```diff
@@ -36,19 +36,29 @@ export async function listWorkflows(
     );
   }
 
+  if (queryParams.inputType === 'query' && !queryParams.query) {
```

Comment:
What would happen if this is not handled this way would it still work and return results?

The reason is that we would need to handle the situation from UI by saying `start by adding search query` and only send request if a text is passed. But why disabling this from the route handler if it is not problematic ?

Reply:
If we did not send a query, it would still return results, yes. You can see this in the old UI but it seems to be a bit flaky.

The initial implementation in the big PR has it just show empty results, but I think we can handle this case separately without any issues. Of course, we can handle it regardless, but I felt making the extra network call was unnecessary since the data returned isn't used at all.

Reply:
Yeah, i agree. 
The points i wanted to make are:
- It can make sense to follow the old UI for having results because no query just means i don't need any filters but there are results.
- On the other side if we decided that the query is a way to request workflows, then we should tell users instead of `No results` `start by adding search query` and handle the logic of disabling the request from UI components not from the route handler.

Reply:
Discussed offline and I've removed it for now, we can re-add it later.